src/oversampling_methods/borderline_smote1/interface_mlj.jl

```diff
@@ -0,0 +1,169 @@
+
+### BorderlineSMOTE1 with MLJ Interface
+
+mutable struct BorderlineSMOTE1{T,R<:Union{Integer,AbstractRNG}, I<:Integer} <: Static
+    m::I
+    k::I
+    ratios::T
+    rng::R
+    try_perserve_type::Bool
+end;
+
+
+
+"""
+Check whether the given model hyperparameters are valid and clean them if necessary. 
+"""
+function MMI.clean!(s::BorderlineSMOTE1)
+    message = ""
+    if s.k < 1
+        throw(ERR_NONPOS_K(s.k))
+    end
+    if s.m < 1
+        throw(ERR_NONPOS_K(s.m))
+    end
+    return message
+end
+
+
+
+
+"""
+Initiate a BorderlineSMOTE1 model with the given hyper-parameters.
+"""
+function BorderlineSMOTE1(;
+    m::Integer = 5,
+    k::Integer = 5,
+    ratios::Union{Nothing,AbstractFloat,Dict{T,<:AbstractFloat}} = 1.0,
+    rng::Union{Integer,AbstractRNG} = default_rng(), try_perserve_type::Bool=true
+) where {T}
+    model = BorderlineSMOTE1(m, k, ratios, rng, try_perserve_type)
+    MMI.clean!(model)
+    return model
+end
+
+"""
+Oversample data X, y using BorderlineSMOTE1
+"""
+function MMI.transform(s::BorderlineSMOTE1, _, X, y)
+    borderline_smote1(X, y; m = s.m, k = s.k, ratios = s.ratios, rng = s.rng, 
+        try_perserve_type = s.try_perserve_type)
+end
+function MMI.transform(s::BorderlineSMOTE1, _, X::AbstractMatrix{<:Real}, y)
+    borderline_smote1(X, y; m = s.m, k = s.k, ratios = s.ratios, rng = s.rng)
+end
+
+
+MMI.metadata_pkg(
+    BorderlineSMOTE1,
+    name = "Imbalance",
+    package_uuid = "c709b415-507b-45b7-9a3d-1767c89fde68",
+    package_url = "https://github.com/JuliaAI/Imbalance.jl",
+    is_pure_julia = true,
+)
+
+MMI.metadata_model(
+    BorderlineSMOTE1,
+    input_scitype = Union{Table(Continuous),AbstractMatrix{Continuous}},
```

Comment:
In the [example](https://github.com/JuliaAI/Imbalance.jl/blob/d152ac3752a86f05431fc6c6bbb588842bbf9794/src/oversampling_methods/borderline_smote1/borderline_smote1.jl#L97) you present in the documentation for the native API, you include categorical inputs. But this input_scitype suggests they are not allowed. Why not?

Reply:
@ablaom it doesn't seem like I do, in the example you link to we have
```
# generate a table and categorical vector accordingly
X, y = generate_imbalanced_data(num_rows, num_continuous_feats; 
                                probs, rng=42)   
```
The keyword argument for categorical columns is not present. Perhaps the comment is misleading? Categorical vector here refers to `y`.


The reason why this only accepts continuous data is that SMOTE only operates on continuous data (line segment logic in continuous space) and BorderlineSMOTE1 is just filtering with SMOTE.

Reply:
You are 100% right. My apologies for my sloppy reading.